src/providers/AccountBalancesProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { createContext, useEffect, useMemo, useRef } from 'react';
+import { useEtherspotBalances, useWalletAddress } from '@etherspot/transaction-kit';
+import { AccountBalance } from '@etherspot/prime-sdk';
+import { sepolia } from 'viem/chains';
+import isEqual from 'lodash/isEqual';
+
+// utils
+import { getAssetBalance, getNativeAssetBalance, supportedChains, usdcOnSepolia } from '../utils/blockchain';
+
+export interface IBalances {
+  [chainId: number]: AccountBalance[];
+}
+
+export interface AccountBalancesContext {
+  listenerRef: React.MutableRefObject<AccountBalancesListenerRef>;
+  data: {
+    balances: IBalances;
+  }
+}
+
+export interface AccountBalancesListenerRef {
+  onBalanceUpdated?: (newBalances: IBalances, prevBalances: IBalances) => void;
+  prevBalances?: IBalances;
+}
+
+export const AccountBalancesContext = createContext<AccountBalancesContext | null>(null);
+
+const AccountBalancesProvider = ({ children }: React.PropsWithChildren) => {
+  const { getAccountBalances } = useEtherspotBalances();
+  const walletAddress = useWalletAddress();
+  const [balances, setBalances] = React.useState<IBalances>({});
+  const listenerRef = useRef<AccountBalancesListenerRef>({});
+
+  useEffect(() => {
+    let expired = false;
+    let timeout: NodeJS.Timeout;
+
+    const refresh = async () => {
+      if (!walletAddress) return;
```

Comment:
Are we warned elsewhere if no walletAddress exists when calling refresh? Just thinking of debugging and monitoring as this would indiate an issue with wallet creation

Reply:
Reason here is not that it's missing, but do not try fetching balances if it's not loaded yet, having logs here doesn't benefit and just results in spam, does it make sense?